src/lxc/confile.c

```diff
@@ -5268,6 +5346,84 @@ static int get_config_net_macvlan_mode(const char *key, char *retv, int inlen,
 	return fulllen;
 }
 
+static int get_config_net_ipvlan_mode(const char *key, char *retv, int inlen,
+				       struct lxc_conf *c, void *data)
+{
+	int len;
+	int fulllen = 0;
+	const char *mode;
+	struct lxc_netdev *netdev = data;
+
+	if (!retv)
+		inlen = 0;
+	else
+		memset(retv, 0, inlen);
+
+	if (!netdev)
+		return -1;
+
+	if (netdev->type != LXC_NET_IPVLAN)
+		return 0;
+
+	switch (netdev->priv.ipvlan_attr.mode) {
+	case IPVLAN_MODE_L3:
+		mode = "l3";
+		break;
+	case IPVLAN_MODE_L3S:
+		mode = "l3s";
+		break;
+	case IPVLAN_MODE_L2:
+		mode = "l2";
+		break;
+	default:
+		mode = "(invalid)";
+		break;
```

Comment:
@brauner that is a good question, the reason I made it like this is because previous network types did it like this, so  wanted to avoid breaking convention. I'm thinking perhaps the reason these "getters" dont return an error for invalid values is because they were intended for use with displaying values and so should gracefully handle invalid values, and that any sort of validation should have been done in the setters. 

For example:

macvlan: https://github.com/lxc/lxc/blob/master/src/lxc/confile.c#L5262-L5263
net type: https://github.com/lxc/lxc/blob/master/src/lxc/confile.c#L3788